test/tbb/test_indexer_node.cpp

```diff
@@ -661,6 +661,81 @@ void test_deduction_guides() {
 
 #endif
 
+#if __TBB_PREVIEW_FLOW_GRAPH_TRY_PUT_AND_WAIT
+void test_try_put_and_wait() {
+    tbb::task_arena arena(1);
+
+    arena.execute([] {
+        tbb::flow::graph g;
+
+        std::vector<int> start_work_items;
+        std::vector<int> processed_items1;
+        std::vector<float> processed_items2;
+        std::vector<int> new_work_items;
+        int wait_message = 10;
+
+        for (int i = 0; i < wait_message; ++i) {
+            start_work_items.emplace_back(i);
+            new_work_items.emplace_back(i + 1 + wait_message);
+        }
+
+        tbb::flow::indexer_node<int, float> indexer(g);
+        using output_type = decltype(indexer)::output_type;
+
+        tbb::flow::function_node<output_type, int> function(g, tbb::flow::serial,
+            [&](output_type tag_msg) noexcept {
+                if (tag_msg.tag() == 0) {
+                    int input = tag_msg.cast_to<int>();
+                    if (input == wait_message) {
+                        for (auto item : new_work_items) {
+                            tbb::flow::input_port<0>(indexer).try_put(item);
+                            tbb::flow::input_port<1>(indexer).try_put(float(item));
+                        }
+                    }
+                    processed_items1.emplace_back(input);
+                } else {
+                    processed_items2.emplace_back(tag_msg.cast_to<float>());
+                }
+                return 0;
+            });
+
+        tbb::flow::make_edge(indexer, function);
+
+        for (auto item : start_work_items) {
+            tbb::flow::input_port<0>(indexer).try_put(item);
+            tbb::flow::input_port<1>(indexer).try_put(float(item));
+        }
+
+        tbb::flow::input_port<0>(indexer).try_put_and_wait(wait_message);
+
+        // Since function is a serial queueing function node, all start_work_items would be stored in a queue
+        // wait_message would be stored at the end of the queue
+        // During the try_put_and_wait call, start_work_items would be processed from the queue in FIFO order
```

Comment:
If it is FIFO that why only wait_msg will be processed?

Reply:
Not only the wait_msg - the order would be start_work_items FIFO then wait_message. The checks are aligned with these expectations. Do you find the comments not clear enough somewhere?